.github/workflows/cypress-workflow.yml

```diff
@@ -0,0 +1,101 @@
+name: E2E Cypress tests
+on:
+  pull_request:
+    branches:
+      - main
+      - 2.*
+  push:
+    branches:
+      - main
+      - 2.*
+env:
+  OPENSEARCH_DASHBOARDS_VERSION: '2.2'
+  OPENSEARCH_VERSION: '2.2.0-SNAPSHOT'
+jobs:
+  tests:
+    name: Run Cypress E2E tests
+    runs-on: ubuntu-latest
+    env:
+      # prevents extra Cypress installation progress messages
+      CI: 1
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Checkout geospatial plugin
+        uses: actions/checkout@v2
+        with:
+          path: geospatial
+          repository: opensearch-project/geospatial
+          ref: '2.2'
+      - name: Run Opensearch with plugin
+        run: |
+          cd geospatial
+          ./gradlew run &
+          timeout 600 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:9200)" != "200" ]]; do sleep 5; done'
+      - name: Checkout Plugin
+        uses: actions/checkout@v2
```

Comment:
I'm seeing the path is specified in another 2 workflows, not sure if it's needed:
ism: https://github.com/opensearch-project/index-management-dashboards-plugin/blob/main/.github/workflows/cypress-workflow.yml#L42-L43
alerting: https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/.github/workflows/cypress-workflow.yml#L46-L47